sidebar-base/src/main/java/com/andrei1058/spigot/sidebar/SidebarManager.java

```diff
@@ -39,31 +39,12 @@ public SidebarManager() throws InstantiationException {
         // load server version support
         String serverVersion = Bukkit.getServer().getClass().getName().split("\\.")[3];
 
-
-        String className = null;
-
-        // latest unmapped version
-        if (serverVersion.equalsIgnoreCase("v1_19_R2")) {
-            className = "com.andrei1058.spigot.sidebar.NineteenProvider";
-        } else if (serverVersion.equalsIgnoreCase("v1_18_R2")) {
-            className = "com.andrei1058.spigot.sidebar.EighteenProvider";
-        } else if (serverVersion.equalsIgnoreCase("v1_17_R1")){
-            className = "com.andrei1058.spigot.sidebar.SeventeenProvider";
-        } else if (serverVersion.equalsIgnoreCase("v1_16_R3")){
-            className = "com.andrei1058.spigot.sidebar.SixteenProvider";
-        } else if (serverVersion.equals("v1_12_R1")) {
-            className = "com.andrei1058.spigot.sidebar.TwelveProvider";
-        } else if (serverVersion.equals("v1_8_R3")) {
-            className = "com.andrei1058.spigot.sidebar.EightProvider";
-        }
-        if (null == className){
-            throw new InstantiationException();
-        }
+        String className = "com.andrei1058.spigot.sidebar." + serverVersion + ".ProviderImpl";
```

Comment:
Does this work with relocation when shading dependencies?

Reply:
> No, but you the old methods also dont work with relocation

It does. We do relocate it in BedWars1058 https://github.com/andrei1058/BedWars1058/blob/master/bedwars-plugin/pom.xml#L307